### DISCUSSION TITLE
Markdown syntax highlighting  not working correctly with Treesitter

### DISCUSSION THREAD

**Discussion by zvrkSam** [LazyVim/LazyVim#2536] [Q&A] [answered]
Right now I am using neovim v.9.5 and I'm relatively new to neovim.
When I load markdown files there is no syntax highlighting for kanagawa theme that I use and for the default themes as well, but there is for tokyonight and cattpuccin.
Looking for a solution i have found that if I uninstall parser for markdown, the defaults are really good, but treesitter has ensure_installed flag for markdown and installs it every time I enter neovim after uninstalling.
I have also considered loading a tokyonight or cattpuccin just for markdown files.
So how do I remove ensure_installed for markdown or load custom theme just for markdown files ?

One more thing, looks like treesitter had some breaking changes I am also willing to wait until this gets resolved.

**Comment by dpetka2001** [comment, marked as answer]
Treesitter has recently changed the symbol names that it uses. Tokyonight has these changes (catppuccin probably as well). It really depends on the maintainer of the theme you use how quickly he will incorporate those changes.

To remove `markdown` from `ensure installed` put the following in some file
```lua
  {
    "nvim-treesitter/nvim-treesitter",
    opts = function(_, opts)
      opts.ensure_installed = vim.tbl_filter(function(i)
        return not vim.tbl_contains({ i }, "markdown")
      end, opts.ensure_installed)
    end,
  },
```

**Reply by zvrkSam** [reply]
Thank you so much for your reply, it works. This is only a quick fix until things get updated
